File: Week_01-数组-链表-堆栈-队列/1281.整数的各位积和之差.py
#
# @lc app=leetcode.cn id=1281 lang=python
#
# [1281] 整数的各位积和之差
#

# @lc code=start
class Solution(object):
    def subtractProductAndSum(self, n):
        """
        :type n: int
        :rtype: int
        """
        add = 0
        mul = 1
        while n > 0:
            digit = n%10
            n //= 10
            add += digit
            mul *= digit
        return mul - add

# @lc code=end

File: Week_01-数组-链表-堆栈-队列/1281.整数的各位积和之差.cpp
/*
 * @lc app=leetcode.cn id=1281 lang=cpp
 *
 * [1281] 整数的各位积和之差
 
 注意审题： n不包括， 如果n包括0，在n=0时，下述方法会得到错误结果1，显然不合理。
 */

// @lc code=start
class Solution {
public:
    int subtractProductAndSum(int n) {
        int product = 1;
        int sum = 0;
            while (n > 0) {
                int remainder = n%10;
                n = n/10;
                product = product * remainder;
                sum += remainder;
            }
        return (product - sum);

    }
};
// @lc code=end

